libopae++/include/opaec++/pvalue.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <opae/properties.h>
+#include <type_traits>
+#include <iostream>
+
+namespace opae
+{
+namespace fpga
+{
+namespace types
+{
+
+template<typename T>
+struct pvalue
+{
+    typedef typename std::conditional<std::is_same<T, char*>::value,
+                                      fpga_result(*)(fpga_properties, T),

```

Comment:
I think I decipher this correctly as a conditional template defintion, specialized for char *?  Nice!

Reply:
yeah, if T is char*, then the getter signature doesn't use T*